src/components/editor/State.js

```diff
@@ -16,22 +16,525 @@ type Props = {
 }
 
 class StateEditor extends Component<Props> {
-  render() {
 
+  renderStateType() {
+    switch (this.props.state.type) {
+      case "Initial":
+        return <Initial {...this.props} />
+      case "Terminal":
+        return <Terminal {...this.props} />
+      case "Simple":
+        return <Simple {...this.props} />
+      case "Guard":
+        return <Guard {...this.props} />
+      case "Delay":
+        return <Delay {...this.props} />
+      case "SetAttribute":
+        return <SetAttribute {...this.props} />
+      case "Counter":
+        return <Counter {...this.props} />
+      case "CallSubmodule":
+        return <CallSubmodule {...this.props} />
+      case "Encounter":
+        return <Encounter {...this.props} />
+      case "EncounterEnd":
+        return <EncounterEnd {...this.props} />
+      case "ConditionOnset":
+        return <ConditionOnset {...this.props} />
+      case "ConditionEnd":
+        return <ConditionEnd {...this.props} />
+      case "AllergyOnset":
+        return <AllergyOnset {...this.props} />
+      case "AllergyEnd":
+        return <AllergyEnd {...this.props} />
+      case "MedicationOrder":
+        return <MedicationOrder {...this.props} />
+      case "MedicationEnd":
+        return <MedicationEnd {...this.props} />
+      case "CarePlanStart":
+        return <CarePlanStart {...this.props} />
+      case "CarePlanEnd":
+        return <CarePlanEnd {...this.props} />
+      case "Procedure":
+        return <Procedure {...this.props} />
+      case "VitalSign":
+        return <VitalSign {...this.props} />
+      case "Observation":
+        return <Observation {...this.props} />
+      case "MultiObservation":
+        return <MultiObservation {...this.props} />
+      case "DiagnosticReport":
+        return <DiagnosticReport {...this.props} />
+      case "Symptom":
+        return <Symptom {...this.props} />
+      case "Death":
+        return <Death {...this.props} />
+      default:
+        return this.props.state.type
+    }
+  }
+
+  render() {
     if(!this.props.state) {
       return null;
     }
     return (
         <div>
-          <RIEInput propName={'name'} value={this.props.state.name} change={this.props.renameNode} />
-          <br />
-          <Transition
```

Comment:
Minor nitpick - wouldn't it have been easier to leave the Transition template in this `render` function than to put it in EVERY state type renderer?

Reply:
You're right. I had left it in my head that I should keep that in each, in case it varied by state, but the transition property should always remain the same. Fixed.